speaking-portal-project/src/types.ts

```diff
@@ -1,52 +1,129 @@
+    __LENGTH
+}
+
+// Eyes Class for updating during runtime
+class Eyes {
+    areClosed : boolean
+    lastBlink: number
+    constructor(areClosed: boolean = false, lastBlink: number = -1) {
+        this.areClosed = areClosed
+        this.lastBlink = lastBlink
+    }
+    close(lastBlink: number) {
+        this.areClosed = true
+        this.lastBlink = lastBlink
+    }
+    open(){
+        this.areClosed = false
+    }
+}
+
+class Mouth {
+    expression: string
+    phoneme: Phoneme
+    constructor(expression: string = '', phoneme: Phoneme = 'A') {
+        this.expression = expression
+        this.phoneme = phoneme
+    }
+}
+
+class Body {
+    currentPose: string
+    lastPoseChange: number
+    constructor(currentPose: string = Pose[0], lastPoseChange: number = -1){
+        this.currentPose = currentPose
+        this.lastPoseChange = lastPoseChange
+    }
+
+    updatePose(poseName: string, lastPoseChange: number){
+        this.currentPose = poseName
+        this.lastPoseChange = lastPoseChange
+    }
+
+}
+
 // Class instead of type to allow for self referencing & updating during runtime
 export class Avatar {
     // Readonly because having the avatar body change would be real funky
     readonly avatar: string
-    blink: boolean
-    phoneme: string
-    arms: string
-    constructor(avatar: string = 'barb', blink: boolean = false, phoneme: string = 'A', arms: string = 'relaxed') {
+    eyes: Eyes
+    mouth: Mouth
+    body: Body
+    constructor(avatar: string = AvatarNames[0], eyes: Eyes = new Eyes(), mouth: Mouth = new Mouth(), body: Body = new Body()) {
         this.avatar = avatar
-        this.blink = blink
-        this.phoneme = phoneme
-        this.arms = arms
+        this.eyes = eyes
+        this.mouth = mouth
+        this.body = body
+    }
+
+    updateMouth(phoneme: Phoneme = this.mouth.phoneme): void {
+        this.mouth.phoneme = phoneme
     }
-    willBlink(currentSec: number, frameDur: string, lastBlink: number) {
-        let eyes: boolean = false
-        if (parseInt(frameDur) < 0.1 && lastBlink != currentSec) {
+
+    updateEyes(currentSec: number, frameDur: number): void {
+        this.eyes.open()
+        if (frameDur < 0.08 && this.eyes.lastBlink != currentSec) {
             let n = Math.random()
-            if ((n <= 0.4 && currentSec % 4 == 0) || (n > 0.4 && n <= 0.8 && currentSec % 3 == 0)) {
-                eyes = true
+            if ((n <= 0.3 && currentSec % 4 == 0) || (n > 0.3 && n <= 0.6 && currentSec % 3 == 0)) {
+                this.eyes.close(currentSec)
             }
         }
-        return eyes
     }
 
-    // Returns the filepath of the image representation the character state
-    generateStateString(duration: string): string {
-        return `file ../images/${this.avatar}/${this.phoneme}_${this.blink ? 'blink' : 'open'}_${
-            this.arms
-        }.png\noutpoint ${duration}\n`
+    // Randomly chooses a new pose that is different from current pose
```

Comment:
I didn't actually keep it from choosing the current pose. I'll do that next week

Reply:
We should maybe put that as an issue to pad those sweet sweet gitstats